buildkite/resource_test_suite_team.go

```diff
@@ -0,0 +1,203 @@
+				Required: true,
+			},
+			"access_level": schema.StringAttribute{
+				Required: true,
+				Validators: []validator.String{
+					stringvalidator.OneOf("MANAGE_AND_READ", "READ_ONLY"),
+				},
+			},
+		},
+	}
+}
+
+func (tst *testSuiteTeamResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var state testSuiteTeamModel
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &state)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	log.Printf("Adding team %s to test suite %s ...", state.TeamID.ValueString(), state.TestSuiteId.ValueString())
+	apiResponse, err := createTestSuiteTeam(
+		tst.client.genqlient,
+		state.TeamID.ValueString(),
+		state.TestSuiteId.ValueString(),
+		SuiteAccessLevels(state.AccessLevel.ValueString()),
+	)
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to create test suite team",
+			fmt.Sprintf("Unable to create test suite team: %s", err.Error()),
+		)
+		return
+	}
+
+	// Set ID and UUID to state
+	state.ID = types.StringValue(apiResponse.TeamSuiteCreate.TeamSuite.Id)
+	state.UUID = types.StringValue(apiResponse.TeamSuiteCreate.TeamSuite.TeamSuiteUuid)
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+func (tst *testSuiteTeamResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state testSuiteTeamModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	log.Printf("Reading test suite team with ID %s ...", state.ID.ValueString())
+	apiResponse, err := getNode(tst.client.genqlient, state.ID.ValueString())
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to read test suite team",
+			fmt.Sprintf("Unable to read test suite team member: %s", err.Error()),
+		)
+		return
+	}
+
+	// Convert fron Node to getNodeTeamMember type
+	if teamSuiteNode, ok := apiResponse.GetNode().(*getNodeNodeTeamSuite); ok {
+		if teamSuiteNode == nil {
+			resp.Diagnostics.AddError(
+				"Unable to get test suite team",
+				"Error getting test suite team: nil response",
+			)
+			return
+		}
+		updateTeamSuiteTeamResource(&state, *teamSuiteNode)
+		resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+	}
+}
+
+func (tst *testSuiteTeamResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+}
+
+func (tst *testSuiteTeamResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var state testSuiteTeamModel
+	var testSuiteTeamAccessLevel string
+
+	resp.Diagnostics.Append(req.Plan.GetAttribute(ctx, path.Root("access_level"), &testSuiteTeamAccessLevel)...)
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+
+	log.Printf("Updating team %s in test suite %s to %s ...", state.TeamID.ValueString(), state.TestSuiteId.ValueString(), testSuiteTeamAccessLevel)
```

Comment:
Same, when will this get output? It doesn't during an `apply` as it's not a Diagnostic.

Reply:
Above ⬆️ - more a TF_LOG thing